test/e2e/scheduling/preemption.go

```diff
@@ -125,4 +125,130 @@ var _ = SIGDescribe("SchedulerPreemption [Serial] [Feature:PodPreemption]", func
+
+			// Apply node label to each node
+			framework.AddOrUpdateLabelOnNode(cs, node.Name, "node", node.Name)
+			framework.ExpectNodeHasLabel(cs, node.Name, "node", node.Name)
+
+			// make the first pod medium priority and the rest low priority.
+			priorityName := lowPriorityClassName
+			if i == 0 {
+				priorityName = mediumPriorityClassName
+			}
+			pods[i] = createPausePod(f, pausePodConfig{
+				Name:              fmt.Sprintf("pod%d-%v", i, priorityName),
+				PriorityClassName: priorityName,
+				Resources: &v1.ResourceRequirements{
+					Requests: podRes,
+				},
+				Affinity: &v1.Affinity{
+					PodAntiAffinity: &v1.PodAntiAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchExpressions: []metav1.LabelSelectorRequirement{
+										{
+											Key:      "service",
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{"blah", "foo"},
+										},
+									},
+								},
+								TopologyKey: "node",
+							},
+						},
+					},
+					NodeAffinity: &v1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								{
+									MatchExpressions: []v1.NodeSelectorRequirement{
+										{
+											Key:      "node",
+											Operator: v1.NodeSelectorOpIn,
+											Values:   []string{node.Name},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			})
+			framework.Logf("Created pod: %v", pods[i].Name)
+		}
+		defer func() { // Remove added labels
+			for i := 0; i < numPods; i++ {
+				framework.RemoveLabelOffNode(cs, nodeList.Items[i].Name, "node")
+			}
+		}()
+
+		By("Wait for pods to be scheduled.")
+		for _, pod := range pods {
+			framework.ExpectNoError(framework.WaitForPodRunningInNamespace(cs, pod))
+		}
+
+		By("Run a high priority pod with node affinity to the first node.")
+		// Create a high priority pod and make sure it is scheduled.
+		runPausePod(f, pausePodConfig{
+			Name:              "preemptor-pod",
+			PriorityClassName: highPriorityClassName,
+			Labels:            map[string]string{"service": "blah"},
+			Affinity: &v1.Affinity{
+				NodeAffinity: &v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{
+									{
+										Key:      "node",
+										Operator: v1.NodeSelectorOpIn,
+										Values:   []string{nodeList.Items[0].Name},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+		// Make sure that the medium priority pod on the first node is preempted to
+		// allow the high priority pod run.
+		preemptedPod, err := cs.CoreV1().Pods(pods[0].Namespace).Get(pods[0].Name, metav1.GetOptions{})
+		podDeleted := (err != nil && errors.IsNotFound(err)) ||
```

Comment:
This logic is a little conflating... more a comment the please change.

Reply:
Sorry, I didn't get the problem. Do you want me to remove the last part of the comment: "to allow the high priority pod run"?